src/v/config/tests/config_store_test.cc

```diff
@@ -358,6 +365,10 @@ SEASTAR_THREAD_TEST_CASE(property_metadata) {
     BOOST_TEST(cfg.nullable_string.is_nullable() == true);
     BOOST_TEST(cfg.nullable_string.is_array() == false);
 
+    BOOST_TEST(cfg.nullable_strings.type_name() == "string");
```

Comment:
Why is the `type_name` string? Shouldn't it be at least `list[string]` or `optional[list[string]]` ?

Reply:
Same reason that vector of string is "string", we want to output this:
```yaml
    "kafka_mtls_principal_mapping_rules": {
      "description": "Principal Mapping Rules for mTLS Authentication on the Kafka API",
      "nullable": true,
      "needs_restart": false,
      "visibility": "user",
      "is_secret": false,
      "type": "array",
      "items": {
        "type": "string"
      }
    },
```